### PR TITLE
Respect filesystem type when mounting mirrors

### PR DESCRIPTION
**Pull request by nonnymoose**
This change reads the filesystem type from /proc/mounts as well and passes that to mount when mounting the mirrors.

This _fixes_ the issue in which devices with ubifs filesystems throw the error
```
mount /dev/ubi0_0->/sbin/.core/mirror/system failed with 15: Block device required
```
about which I think I complained at you a few months ago.

It should also improve compatibility with other strange filesystem types, but I have no way of testing that.

### CODE REVIEW

File: native/jni/core/bootstages.c [hunk 1]
```diff
@@ -802,6 +804,7 @@ void late_start(int client) {
 	// Allocate buffer
 	if (buf == NULL) buf = xmalloc(PATH_MAX);
 	if (buf2 == NULL) buf2 = xmalloc(PATH_MAX);
+	if (buf3 == NULL) buf3 = xmalloc(PATH_MAX);
```
Comment (nonnymoose) [inline native/jni/core/bootstages.c]: There should probably be a saner default for allocating the filesystem buffer, but I'm not sure what it is.

**Comment by npes87184** [comment]
Hi, @nonnymoose 

buf3 will memory leak.
Remember to free buf3 like buf and buf2 do in [Line 841](https://github.com/topjohnwu/Magisk/blob/master/native/jni/core/bootstages.c#L841).

Thanks.

**Comment by nonnymoose** [comment]
@npes87184 Thanks for your suggestions! I have fixed the memory leak, but bricked my device in the process :(, so it should be an hour (or three) before I can test it and commit.

**Comment by nonnymoose** [comment]
@npes87184 OK, I have done that. Do you have an idea of what the maximum fs_type size should be?
The longest filesystem supported by my Ubuntu machine is 11 characters.

**Comment by nonnymoose** [comment]
@npes87184 I agree with your suggestions. The maximum length filesystem type I could find was still 11, but it can't hurt to be _really, really_ careful, so 64 seems good to me.

**Comment by npes87184** [comment]
Hi, @nonnymoose 

Well, the static allocated I meant is `static char buf_fs_type[64] = {0}`.
But, I don't know whether use static allocated is a good idea or not. This question is left to @topjohnwu.

Regards!

**Comment by nonnymoose** [comment]
Sorry about that. I'm not used to c and tend to get terminology wrong.

**Comment by nonnymoose** [comment]
@topjohnwu Thanks so much!

**Pull request closed without merging**